### PR TITLE
Improve handling of arguments

### PR DESCRIPTION
**Pull request by dekrain**

### CODE REVIEW

File: src/driver/args.c [hunk 1]
```diff
@@ -110,100 +105,116 @@ void yf_parse_args(int argc, char ** argv, struct yf_args * args) {
...
         }
 
-        if (STREQ(arg, "-h") || STREQ(arg, "--help")) {
-            yf_check_action(args, YF_HELP);
-            continue;
-        }
+        if (arg[0] == '-') {
+            ++arg;
+            if (!arg[1]) {
+                if (arg[0] == 'h') {
```
Comment (adamhutchings) [inline src/driver/args.c:111]: But what if the user calls with `yfc -hey-this-is-invalid`?

Comment (dekrain) [reply src/driver/args.c:111]: Then `arg[1]` wouldn't be `0` :)

**Comment by dekrain** [comment]
The comment in help message
```
yfc <file1> <file2> ...: compile and link the given files, up to 16.
```
should be updated not to include the file limit, but I don't want to run another CI job :C

File: src/driver/args.c [hunk 2]
```diff
@@ -110,100 +105,116 @@ void yf_parse_args(int argc, char ** argv, struct yf_args * args) {
...
             }
-            continue;
-        }
 
-        if (STREQ(arg, "-compiler-type")) {
-            want_compiler_type = true;
-            if (i + 1 == argc) {
-                yf_set_error(args);
-                return;
+            if (STREQ(arg, "version")) {
```
Comment (adamhutchings) [inline src/driver/args.c:130]: The string is `--version`, not `-version`.

Comment (dekrain) [reply src/driver/args.c:130]: I made it so that long options can be specified with either one dash or two.

**Review by adamhutchings — APPROVED** [review]
Looks good! Would you allow me to merge mine first, though?

File: tests/.gitignore [hunk 3]
```diff
@@ -0,0 +1,2 @@
+*.o
```
Comment (adamhutchings) [inline tests/.gitignore:1]: Good idea.

**Review by adamhutchings — APPROVED** [review]

**Pull request merged**